odxtools/compumethods/texttablecompumethod.py

```diff
@@ -28,25 +28,30 @@ def __post_init__(self) -> None:
     def category(self) -> CompuMethodCategory:
         return "TEXTTABLE"
 
-    def _get_scales(self):
+    def _get_scales(self) -> List[CompuScale]:
         scales = list(self.internal_to_phys)
         if self.compu_default_value:
             # Default is last, since it's a fallback
             scales.append(self.compu_default_value)
         return scales
 
-    def convert_physical_to_internal(self, physical_value):
-        scale: CompuScale = next(
-            filter(lambda scale: scale.compu_const == physical_value, self._get_scales()), None)
-        if scale is not None:
-            res = (
-                scale.compu_inverse_value
-                if scale.compu_inverse_value is not None else scale.lower_limit.value)
+    def convert_physical_to_internal(self, physical_value: AtomicOdxType) -> AtomicOdxType:
+        scales = [x for x in self._get_scales() if x.compu_const == physical_value]
+        if scales:
+            inverse_value: Optional[AtomicOdxType] = scales[0].compu_inverse_value
+            if inverse_value is not None:
+                res = inverse_value
+            elif scales[0].lower_limit is not None:
+                res = scales[0].lower_limit.value
+            else:
+                res = None
+
             odxassert(self.internal_type.isinstance(res))
-            return res
-        raise EncodeError(f"Texttable compu method could not encode '{physical_value}'.")
+            return res  # type: ignore[return-value]
```

Comment:
Why are you ignoring return value type?

Reply:
mypy produces the following error:
```
odxtools/compumethods/texttablecompumethod.py:50: error: Incompatible return value type (got "str | int | float | bytes | None", expected "str | int | float | bytes")  [return-value]
Found 1 error in 1 file (checked 10 source files)
```

since line 49 ensures that res is not `None`, it is a false positive... (I don't think there's a practical way of properly conferring this fact to mypy.)

Reply:
```suggestion
            return return odxrequire(res)
```

Reply:
problem caused by line 47, `res = None`

Reply:
nice catch. fixed, thanks!